File: programovani/ulohy/files/modelovani-obrazcu-iii/pyramida-obracene.cpp
#include <iostream>
using namespace std;


int main() {

    // Deklarace a inicializace
    const int vel = 4; // Inicializace velikosti pyramidy
    int pom = 0; // Inicializace pomocne promenne

    // Vnejsi cyklus pro radky
    for(int i = vel; i >= 1; i--) {
        
        // Vnitrni cyklus pro jednotlive hvezdicky
        for(int j = 1; j <= vel + i - 1; j++) {

            // Pro zacatecni prazdne bunky
            if(pom >= j) {

                // Vypiseme pouze mezeru
                cout << "  ";

            }

            // Pokud jsme jiz v limitu
            else {

                // Vypiseme hvezdicku
                cout << "* ";

            }

        }            

        // Vypiseme novou radku
        cout << endl;

        // Pridame sloupec pocitadlu
        pom++;

    }

}

File: programovani/ulohy/files/modelovani-obrazcu-iii/pyramida-obracene.py
# Deklarace a inicializace
vel = 4 # Inicializace velikosti pyramidy
pom = 0 # Inicializace pomocne promenne

# Vnejsi cyklus pro radky
for i in range(vel, 0, -1):
    
    # Vnitrni cyklus pro jednotlive hvezdicky
    for j in range(1, vel + i):

        # Pro zacatecni prazdne bunky
        if(pom >= j):

            # Vypiseme mezeru
            print("  ", sep="", end="")

        # Pokud jsme jiz v limitu
        else:

            # Vypiseme hvezdicku
            print("* ", sep="", end="")           

    # Vypiseme novou radku
    print(end="\n")

    # Pridame sloupec pocitadlu
    pom += 1